resources/assets/js/components/H2Components/Dialog.tsx

```diff
@@ -123,39 +121,124 @@ const Trigger: React.FunctionComponent<TriggerProps> = (props) => {
 };
 
 interface DialogComposition {
-  Actions: React.FunctionComponent<GeneralProps>;
-  ActionBtn: React.FunctionComponent<GeneralBtnProps>;
-  Content: React.FunctionComponent<GeneralProps>;
-  Header: React.FunctionComponent<GeneralProps>;
-  Title: React.FunctionComponent<GeneralProps>;
+  Actions: React.FunctionComponent<ActionsProps>;
+  ActionBtn: React.FunctionComponent<ActionBtnProps>;
+  Content: React.FunctionComponent<ContentProps>;
+  Header: React.FunctionComponent<HeaderProps>;
+  Title: React.FunctionComponent<TitleProps>;
   Trigger: React.FunctionComponent<TriggerProps>;
 }
 
 const Dialog: React.FunctionComponent<DialogContext> & DialogComposition = (
   props,
 ) => {
-  const { id, className, children, ...rest } = props;
-  const ref = React.useRef(null);
+  const { id, isVisible, className, children, closeDialog, ...rest } = props;
+  // Focus the first focusable element when the modal becomes visible.
+  const dialogRef = React.useRef<HTMLDivElement | null>(null);
+  React.useEffect(() => {
+    if (isVisible && dialogRef.current) {
+      const focusableModalElements = getTabList(dialogRef.current);
+      if (focusableModalElements.length > 0) {
+        const firstElement = focusableModalElements[0] as HTMLElement;
+        firstElement.focus();
+      }
+    }
+  }, [isVisible]);
+
+  const handleKeyUp = React.useCallback(
+    (event) => {
+      switch (event.key) {
+        case "Escape":
+          closeDialog();
+          focusOnElement(`[data-h2-dialog-trigger=${id}]`);
```

Comment:
This isn't quite a full solution - because we could potentially have multiple triggers for one dialog, and it wouldn't necessarily return to the right one.

Reply:
I'm still not sure how to handle this best.

Reply:
But I would consider fixing this to be the purview of #5463

Reply:
Good point, not sure atm either.